SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp

```diff
@@ -276,30 +299,18 @@ Interest InterestFactory::CreateActorInterest() const
 	// e.g. Handover, OwnerOnly, etc.
 	NewQuery.FullSnapshotResult = true;
 
-	ComponentInterest NewComponentInterest;
-	NewComponentInterest.Queries.Add(NewQuery);
-
-	// Server Interest
-	NewInterest.ComponentInterestMap.Add(SpatialConstants::POSITION_COMPONENT_ID, NewComponentInterest);
-
-	return NewInterest;
+	AddComponentQueryPairToInterestComponent(OutInterest, SpatialConstants::POSITION_COMPONENT_ID, NewQuery);
 }
 
-Interest InterestFactory::CreatePlayerOwnedActorInterest() const
+void InterestFactory::AddPlayerControllerActorInterest(Interest& OutInterest) const
 {
 	const USpatialGDKSettings* SpatialGDKSettings = GetDefault<USpatialGDKSettings>();
 
-	QueryConstraint SystemConstraints = CreateSystemDefinedConstraints();
-
-	// Servers only need the defined constraints
-	Query ServerQuery;
-	ServerQuery.Constraint = SystemConstraints;
-	ServerQuery.FullSnapshotResult = true;
+	const Worker_ComponentId ClientEndpointComponentId = SpatialConstants::GetClientAuthorityComponent(SpatialGDKSettings->bUseRPCRingBuffers);
```

Comment:
Nit: Guess this is strange that this is set a long way from where it is used (line 340)?

Reply:
set it at the top somewhat arbitrarily as it is used throughout. Moved